doc/Makefile

```diff
@@ -104,7 +104,7 @@ html: xmlgen $(SPHINXCONFIG)/conf.py $(ANCHORCHECK) $(MATHJAX) $(POLYFILL)
 	@rm -rf html/JPG
 	@cp -r src/PDF html/PDF
 	@rm -rf html/PDF/.[sg]*
-	@echo "Build finished. The HTML pages are in doc/html."
+	@echo "Build finished. The HTML pages are in doc/html. Documentation starts with doc/html/Manual.html"
```

Comment:
I have not reverted this change. 

If we are going to generate a 0-length index.html, which is the file that is normally the root of any HTML fileset, then I believe we should provide some clue about which other file is the root for this fileset.

Double-clicking the empty index.html, which is the natural next step for a newbie after the doc build finishes, generates a scary, confusing error message in some common browsers (see image for Chrome) or an apparent hang in others.
![image](https://user-images.githubusercontent.com/6098961/92064737-590c3300-ed5b-11ea-9689-88b8fa61a4e2.png)

Reply:
the problem is that you are using an unsuitable operating system (or rather file system). You are seeing an empty file, because of the limitations of your file system. The command in the makefile does not create an empty file but a symbolic link. It is supposed to be like that and your PR will not be accepted without undoing the change as I requested.

if you must work on a windows machine, you should consider using the windows subsystem for Linux and run within a proper programming environment. https://docs.lammps.org/Howto_wsl.html

Reply:
@akohlmey My build and test environments are exclusively the Ubuntu subsystem for Windows. Why do you think otherwise?

It's true I use the git bash environment for working with source code control, so my files have windows line endings locally. But I would think any developer would need that to use a Windows-based IDE. And, of course, my browsers are running in Windows.

This is exactly the same environment I used when I worked on with docs builds previously.

It being the end of the day before a long weekend, I'll simply close this pull request and consider regrouping next week.

Reply:
> @akohlmey My build and test environments are exclusively the Ubuntu subsystem for Windows. Why do you think otherwise?

i am guessing. I only see the pathname containing a "C:". That is usually an indication of somebody working on windows.
The fact that your webbrowser is confused is Windows thing or due to using a file system not supporting symbolic links (you can run into that situation also on a MacOS or Linux box, if you mount, for example an NTFS file system).

> It's true I use the git bash environment for working with source code control, so my files have windows line endings locally. But I would think any developer would need that to use a Windows-based IDE. And, of course, my browsers are running in Windows.

I disagree with this statement. It is certainly not true for most of the LAMMPS developers I know. Large parts of (scientific) software have been written without an IDE and without Windows.  I do my programming on a "true" Linux machine with a mix of Emacs and Vim (each is better at some tasks than the other and I try to take advantage of that). Even the windows binary packages we provide are compiled on a Linux machine with a cross-compiler. Please also keep in mind that LAMMPS is particularly popular on HPC environments, and those run almost exclusively on Linux machines these days. Same for most cloud computing services.

I have been working on Linux machines (and before that on different terminals connected to Unix and VMS servers) since college (and that was quite a long time ago). There is an unparalleled degree of consistency and continuity when working like that. I stopped using Windows when I found out how much more productive I can be on a Linux/Unix machine. These days I have one tiny machine that is dual boot and that may be booted into Windows perhaps once every 2-3 months (and then I get punished for that with endless downloads of updates and reboots).

> 
> This is exactly the same environment I used when I worked on with docs builds previously.
> 
> It being the end of the day before a long weekend, I'll simply close this pull request and consider regrouping next week.

Reply:
**** attempt begins to inject some humor in a discussion that seems to be getting heated ****
@akohlmey I know from long, long experience that I cannot win a religious discussion with either a Vim or an Emacs developer. If you are both, I am doubly defeated. Please, take pity on me.
**** attempt ends

Seriously, I'm NOT arguing FOR the use of Windows in any way, shape, or form. I agree completely Linux is the way to go for serious developers if you have a choice.  I do understand that the vast majority of lammps users/devs are on either Linux or OsX. But I make my living with Windows, so it's what I have to use for open source contributions. And recent work done on the docs (e.g. @rbberger #2247) had reinforced my sense that Windows+Ubuntu subsystem was an accepted contributor environment.

The only point I was trying to make is that I believe the tools environment I now have, doing all the builds and tests in the Ubuntu subsystem but using a visual editor (so not the full features of an ide, simply a capable visual editor), is a typical combination for us poor-benighted-brethren-who-must-use-Windows. 

I'm more than willing to be wrong about a change I've proposed. I've already closed this PR and will reconsider my view next week. I truly value the role you play as guardian for this enormous code base. I simply don't think it's the fault of my dev environment.

Reply:
> **** attempt begins to inject some humor in a discussion that seems to be getting heated ****

if you consider this a heated discussion, you haven't seen a heated discussion. you should look up some of the arguments I've been getting into when I wasn't carrying as much responsibility for LAMMPS development and user support as I do now.

> @akohlmey I know from long, long experience that I cannot win a religious discussion with either a Vim or an Emacs developer. If you are both, I am doubly defeated. Please, take pity on me.

you have the wrong impression. i am not religious about this. i would not be using both vi and emacs if I was. ;-)
i have no need to win anything here. I have taught many workshops and classes with scientists writing scientific software and in my observation people using IDEs are the exception not the rule. 

> **** attempt ends
> 
> Seriously, I'm NOT arguing FOR the use of Windows in any way, shape, or form. I agree completely Linux is the way to go for serious developers if you have a choice. I do understand that the vast majority of lammps users/devs are on either Linux or OsX. But I make my living with Windows, so it's what I have to use for open source contributions. And recent work done on the docs (e.g. @rbberger #2247) had reinforced my sense that Windows+Ubuntu subsystem was an accepted contributor environment.

This is all fine. What rubs me the wrong way are statements that implicitly make the assumption that working on windows is the norm and everything has to be **fully** compatible with that (the "fully" is the key here). We try to accommodate, but there are limits. Compiling LAMMPS on Windows with the native (i.e. Visual C++) compilers is one of them (it would require a significant porting effort and will only happen if we run out of more interesting projects or there is a volunteer that will do the work in a reasonably clean fashion). It can be made to work and kept working for Cygwin and MinGW (native or via cross compiler) and that we try to keep working (since we do recognized there is a significant user community running LAMMPS on windows). Building the documentation with the makefile in the doc folder is only expected to work on Linux and it appears the building is working for you as well. So that is all fine and that what we aim for.

> I'm more than willing to be wrong about a change I've proposed. I've already closed this PR and will reconsider my proposed change next week. I truly value the role you play as guardian for this enormous code base. I simply don't think it's the fault of my dev environment.

As I already stated, the fact that symbolic links appear as empty files to native windows applications **is** a limitation of your environment and thus not something that we need to address in the source code or build system side. Symbolic links are a requirement for how we build the LAMMPS tar packages for download and how the manual is hosted on the web server(s). Please note that this is for building the manual and not for the LAMMPS application itself. There platform dependent code and features are limited to the minimum and for the most part there is either some kind of abstraction or equivalent implementations or dummy functionality the informs about a feature not being able (like you cannot use multi-threading without compiling it with OpenMP support enabled).

Bottom line, developing on windows is fine for as long as one is aware of the limitations and is careful to not let windows limitations (e.g. lack of permissions, CR-LF line endings) disrupt the development process and cause extra work. Using the WSL is by far preferred over using windows natively because it has less limitations. But even then we have accommodated people in the past when they reported issues (e.g. about the native file system being unable to handle a file called '...') when it would be easy to do and not create issues otherwise. And that is why we need to keep the command producing the symbolic link in the html folder.

Reply:
Just to close out the discussion, I will lock this thread now. Even if a bit verbose, it is good to have had a discussion on the boundaries of what is expected to work how and what will be accommodated.  If you feel we need to discuss some more, we can do so via e-mail or other personal communication.